src/decimal.rs

```diff
@@ -636,6 +636,26 @@ impl FromPrimitive for Decimal {
 }
 
 impl ToPrimitive for Decimal {
+
+    fn to_f64(&self) -> Option<f64> {
+        if self.scale() == 0 {
+            let bytes = self.unsigned_bytes_le();
+            let sign;
+            if self.is_negative() {
+                sign = Minus;
+            } else {
+                sign = Plus;
+            }
+            
+            BigInt::from_bytes_le(sign, &bytes[..]).to_f64()
+        } else {
+            match self.to_string().parse::<f64>() {
```

Comment:
I wonder if I need to look at the overflow cases in further detail going forward? Just thinking that at the moment it'd return `None` which may result in unintended functionality. This code as it stands is consistent with the rest of the library so should be ok for now.

Reply:
What overflows are you thinking of? Decimal::max_value() does not overflow f64. It does result in a loss of precision, but I think that's OK.